backend/api_handler.py

```diff
@@ -131,56 +116,64 @@ def handler(event, context):
         if 'user_id' in event['requestContext']['authorizer']:
             user_id = event['requestContext']['authorizer']['user_id']
         log.debug('username is %s', username)
-        try:
-            groups = []
-            if os.environ.get('custom_auth', None):
-                groups.extend(get_custom_groups(user_id))
-            else:
-                groups.extend(get_cognito_groups(claims))
-            log.debug('groups are %s', ','.join(groups))
-            with ENGINE.scoped_session() as session:
-                for group in groups:
-                    policy = TenantPolicyService.find_tenant_policy(session, group, TenantPolicyService.TENANT_NAME)
-                    if not policy:
-                        print(f'No policy found for Team {group}. Attaching TENANT_ALL permissions')
-                        TenantPolicyService.attach_group_tenant_policy(
-                            session=session,
-                            group=group,
-                            permissions=TENANT_ALL,
-                            tenant_name=TenantPolicyService.TENANT_NAME,
-                        )
 
-        except Exception as e:
-            print(f'Error managing groups due to: {e}')
-            groups = []
+        groups: list = extract_groups_and_attach_tenant_policy(user_id=user_id, claims=claims)
 
+        # Set Context
         set_context(RequestContext(ENGINE, username, groups, user_id))
-
         app_context = {
             'engine': ENGINE,
             'username': username,
             'groups': groups,
             'schema': SCHEMA,
         }
 
-        # Determine if there are any Operations that Require ReAuth From SSM Parameter
-        try:
-            reauth_apis = ParameterStoreManager.get_parameter_value(
-                region=os.getenv('AWS_REGION', 'eu-west-1'), parameter_path=f'/dataall/{ENVNAME}/reauth/apis'
-            ).split(',')
-        except Exception:
-            log.info('No ReAuth APIs Found in SSM')
-            reauth_apis = None
+        query = json.loads(event.get('body'))
+
+        isBlockedResponse = validate_and_block_if_maintenance_window(query=query, groups=groups)
```

Comment:
I am not sure is Blocked returns anything

Reply:
same for `check_reauth`

Reply:
Hi @dlpzx , `validate_and_block_if_maintenance_window` and `check_reauth` both return responses. 

`validate_and_block_if_maintenance_window` returns response if its unauthorized for that type of maintenance window


`check_reauth` -> returns response if there is re_auth operation.